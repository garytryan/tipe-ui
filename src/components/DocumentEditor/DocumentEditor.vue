<template>
  <div :data-tipe-ui="$options.name">
    <tipe-document-block-list
      :blocks="blocks"
      :options="options"
    />
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import blockOptionShape from '@/types/BlockOption'
import documentBlockShape from '@/types/DocumentBlock'
import TipeDocumentBlockList from './DocumentBlockList'

export default {
  name: 'TipeDocumentEditor',
  components: {
    TipeDocumentBlockList
  },
  props: {
    options: vueTypes.arrayOf(vueTypes.shape(blockOptionShape)),
    blocks: vueTypes.arrayOf(vueTypes.shape(documentBlockShape))
  }
}
</script>
