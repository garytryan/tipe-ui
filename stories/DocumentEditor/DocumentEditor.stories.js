import { storiesOf } from '@storybook/vue'
import TipeDocumentEditor from '@/components/DocumentEditor'
import documentBlockMock from '@/mocks/DocumentBlock'
import blockOption from '@/mocks/BlockOption'
import createManyMocks from '@/mocks/createManyMocks'

const blocks = [
  documentBlockMock({
    value: true,
    status: '',
    waiting: false,
    disabled: false
  }),
  documentBlockMock({
    value: true,
    status: 'success',
    waiting: false,
    disabled: false
  }),
  documentBlockMock({
    value: true,
    status: 'warning',
    waiting: false,
    disabled: false
  }),
  documentBlockMock({
    value: true,
    status: 'error',
    waiting: false,
    disabled: false
  }),
  documentBlockMock({
    value: true,
    status: '',
    waiting: true,
    disabled: false
  }),
  documentBlockMock({
    value: true,
    status: '',
    waiting: false,
    disabled: true
  })
]

storiesOf('DocumentEditor', module)
  .add('empty', () => ({
    components: { TipeDocumentEditor },
    data() {
      return {
        styleObj: {
          height: '35rem',
          width: '35rem'
        },
        options: createManyMocks(blockOption, 3),
        blocks
      }
    },
    template: `
      <div :style="styleObj">
        <tipe-document-editor
          :options="options"
        />
      </div>`
  }))
  .add('with blocks', () => ({
    components: { TipeDocumentEditor },
    data() {
      return {
        styleObj: {
          height: '35rem',
          width: '35rem'
        },
        options: createManyMocks(blockOption, 3),
        blocks
      }
    },
    template: `
      <div :style="styleObj">
        <tipe-document-editor
          :blocks="blocks"
          :options="options"
        />
      </div>`
  }))
