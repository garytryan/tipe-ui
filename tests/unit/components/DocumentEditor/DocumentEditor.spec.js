import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import DocumentEditor from '@/components/DocumentEditor'
import documentBlockFixture from '@/fixtures/DocumentBlock'
import documentBlockMock from '@/mocks/DocumentBlock'

describe('DocumentBlock', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const propsData = { block: documentBlockFixture() }
      const wrapper = shallowMount(DocumentEditor, { propsData })

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const propsData = { block: documentBlockMock() }
      const wrapper = shallowMount(DocumentEditor, { propsData })

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeDocumentEditor')
    })
  })
})
