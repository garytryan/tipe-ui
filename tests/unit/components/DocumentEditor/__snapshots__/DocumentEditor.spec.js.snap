// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentBlock <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeDocumentEditor">
  <TipeDocumentBlockList-stub></TipeDocumentBlockList-stub>
</div>
`;
